src/hex.cpp

```diff
@@ -86,11 +86,8 @@ namespace libtorrent
 	std::string to_hex(span<char const> in)
 	{
 		std::string ret;
-		if (!in.empty())
-		{
-			ret.resize(in.size() * 2);
-			to_hex(in.data(), in.size(), &ret[0]);
-		}
+		ret.resize(in.size() * 2);
+		to_hex(in.data(), in.size(), &ret[0]);
```

Comment:
oh, I get it now.. ``data()`` returns a pointer to an immutable string (until C++17). So for now we still need ``&ret[0]`` *and* the check for empty string.